samples/snippets/src/test/java/com/example/datastore/RegionalEndpointIT.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.datastore;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Key;
+import com.rule.SystemsOutRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for quickstart sample. */
+@RunWith(JUnit4.class)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public class RegionalEndpointIT {
+
+  @Rule public final SystemsOutRule systemsOutRule = new SystemsOutRule();
+
+  private static final void deleteTestEntity() {
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    String kind = "Task";
+    String name = "sampletask1";
+    Key taskKey = datastore.newKeyFactory().setKind(kind).newKey(name);
+    datastore.delete(taskKey);
+  }
+
+  @Before
+  public void setUp() {
+    deleteTestEntity();
+  }
+
+  @After
+  public void tearDown() {
+    System.setOut(null);
+    deleteTestEntity();
+  }
+
+  @Test
+  public void testQuickstart() throws Exception {
+    QuickstartSample.main();
```

Comment:
This won't work as is since you've hardcoded the project id. I would recommend getting the `GOOGLE_CLOUD_PROJECT` environment variable (set here https://github.com/googleapis/java-datastore/blob/main/.kokoro/presubmit/samples.cfg#L20-L23) and passing it through as a parameter. Here's an example in Bigtable which does something similar: https://github.com/googleapis/java-bigtable/blob/main/samples/snippets/src/test/java/com/example/bigtable/QuickstartTest.java#L37, https://github.com/googleapis/java-bigtable/blob/4e979cb57e5fd8887d5ea413169d80be75e6a621/samples/snippets/src/test/java/com/example/bigtable/BigtableBaseTest.java#L63